### PR TITLE
[flang] Implement GETUID and GETGID intrinsics

### PR DESCRIPTION
**Pull request by DavidTruby**
GETUID and GETGID are non-standard intrinsics supported by a number of
other Fortran compilers. On supported platforms these intrinsics simply
call the POSIX getuid() and getgid() functions and return the result.
The only platform we support that does not have these is Windows.

Windows does not have the same concept of UIDs and GIDs, so on Windows
we issue a warning indicating this and return 1 from both functions.

Co-authored-by: Yi Wu <yi.wu2@arm.com>

### CODE REVIEW

File: flang/lib/Semantics/check-call.cpp [hunk 1]
```diff
@@ -2020,6 +2020,20 @@ bool CheckPPCIntrinsic(const Symbol &generic, const Symbol &specific,
   return false;
 }
 
+bool CheckWindowsIntrinsic(
```
Comment (tblah) [inline flang/lib/Semantics/check-call.cpp:2023]: This is a good idea. Are these the only two intrinsics which are not supported on Windows? There is at least ACCESS which is also unsupported.

Comment (DavidTruby) [reply flang/lib/Semantics/check-call.cpp:2023]: We should probably do a check through to see which intrinsics aren't working on Windows and add them all here. That should be a separate patch though

Comment (tblah) [reply flang/lib/Semantics/check-call.cpp:2023]: Agreed. Please add a TODO here

**Comment by klausler** [comment]
Why do these have to be intrinsics rather than just extensions in flang/runtime/extensions.{h, cpp}, like `sleep`?

**Comment by DavidTruby** [comment]
getuid and getpid return unsigned integers so I think we have to add a convert between that and a normal Fortran integer after the call.

**Comment by DavidTruby** [comment]
Just to clarify what I was trying to say:
for compatibility with gfortran we need to return the value as the default integer kind, which I don't think we can do from the runtime as we don't know what the user has set the default integer to at that point.
Compatibility on this particular point might not matter though I suppose.

**Comment by DavidTruby** [comment]
Yes, I believe so. That does mean we need to have this as an intrinsic and not an extension though, so I was asking your approval to continue it this way.

**Comment by klausler** [comment]
> Yes, I believe so. That does mean we need to have this as an intrinsic and not an extension though, so I was asking your approval to continue it this way.

Do they work with GNU when one uses its command-line options to change the default kind of integer?  If not, then you don't need to make that work here, either.  But if they do, then yes, support as an intrinsic function will be needed.

**Review by Leporacanthicus — APPROVED** [review]
LGTM

**Comment by abidh** [comment]
The `Semantics/windows.f90` test fails for me. Can you kindly add a `required` line or something so that it does not break for people who don't build `aarch64` target.

**Pull request merged**